homeassistant/components/tuya/climate.py

```diff
@@ -231,6 +231,11 @@ def __init__(
                 description.switch_only_hvac_mode,
             ]
 
+        # determine HVAC mode DPCode to use POWER_SWITCH and not SWITCH if available
+        self._hvac_mode_dp_name = DPCode.SWITCH
+        if DPCode.POWER_SWITCH in self.device.function:
```

Comment:
This ain't right, we have helper methods for this, please use the find_dpcode methods to find them.